contextx/daemon_test.go

```diff
@@ -0,0 +1,66 @@
+package contextx_test
+
+import (
+	"context"
+	"math/rand"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/socialpoint-labs/bsk/contextx"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestRunDaemon(t *testing.T) {
+	ch := make(chan struct{})
+
+	runner := func() contextx.RunnerFunc {
+		return func(ctx context.Context) {
+			ch <- struct{}{}
+		}
+	}
+
+	go contextx.RunDaemon(runner(), runner())
+
+	<-ch
+	<-ch
```

Comment:
consider adding a timeout here, otherwise we'll have to wait 10 minutes (default test timeout)

Reply:
I tackled the same thing in the test of the metrics recorder, see: https://github.com/socialpoint-labs/bsk/pull/38/files#diff-5aac0f7f200b99ed402503d8e168c5d5R196

If we agree on that I will take the same approach in this PR.